### PR TITLE
HDDS-9232. Write performance degradation.

### PR DESCRIPTION
**Pull request by szetszwo**
## What changes were proposed in this pull request?

The write performance degradation is due to the ratis conf `raft.server.log.appender.wait-time.min` default value 10ms.  We will set it to 1ms as default for OM, SCM and datanode.

## What is the link to the Apache JIRA

HDDS-9232

## How was this patch tested?

New unit tests.

### CODE REVIEW

**Comment by kerneltime** [comment]
Moving the discussion to the jira for now, till the design issues are resolved and we can jump to the code review.

**Comment by szetszwo** [comment]
Build failure due to the following error.
```
[INFO] --- aspectj:1.14.0:compile (default) @ ozone-manager ---
[INFO] Showing AJC message detail for messages of types: [error, warning, fail]
[ERROR] Internal compiler error: java.lang.Exception: java.lang.NullPointerException at org.aspectj.org.eclipse.jdt.internal.compiler.apt.dispatch.RoundDispatcher.handleProcessor(RoundDispatcher.java:172)
	/Users/szetszwo/ozone/ozone-fork/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/audit/package-info.java:0
(no source information available)
```
After updated `aspectj.version` to 1.9.20.1, the error goes away.

**Comment by szetszwo** [comment]
This is an example of the build failure:
https://github.com/apache/ozone/actions/runs/6254665078/job/17013840228?pr=5338#step:6:2126

**Comment by szetszwo** [comment]
> After updated `aspectj.version` to 1.9.20.1, the error goes away.

Indeed, upgrading `aspectj.version` did not fix the error.  Adding `OzoneManagerRatisServerConfig extends RatisServerConfig` caused the aspectj compilation NPE.

**Comment by kerneltime** [comment]
Does this mean any thread writing to a ratis pipeline will not exceed 1000 ops per second? Can you explain why sleeping 1 ms is desirable?

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/ScmRatisServerConfig.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+/**
+ * SCM Ratis Server config.
+ */
+@ConfigGroup(prefix = ScmConfigKeys.OZONE_SCM_HA_PREFIX
+    + "." + RaftServerConfigKeys.PREFIX)
+public class ScmRatisServerConfig {
+  /** @see RaftServerConfigKeys.Log.Appender#WAIT_TIME_MIN_KEY */
+  @Config(key = "log.appender.wait-time.min",
+      defaultValue = "1ms",
```
Comment (kerneltime) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/ScmRatisServerConfig.java:38]: Why not 0?

**Comment by szetszwo** [comment]
> Does this mean any thread writing to a ratis pipeline will not exceed 1000 ops per second?

No, a single `appendEntries` call can send multiple log entries (i.e. multiple transactions).  As an example, see the diagram in the description of [RATIS-1886 ](https://issues.apache.org/jira/browse/RATIS-1886).  Even for 10ms, we can get ~3000 ops per second.

> Can you explain why sleeping 1 ms is desirable?

Quite a few people have tested with 1ms and 0ms.  They found no difference in performance. (@duongkame, do you agree?)

On the other hand, 1ms does protect against the problem of sending `appendEntries` too frequently.

**Comment by szetszwo** [comment]
> On the other hand, 1ms does protect against the problem of sending appendEntries too frequently.

We only observe the problem in Datanodes.  Let's change the default of OM and SCM to 0ms.

**Review by duongkame — APPROVED** [review]
LGTM, +1. Please address the failed CI before merging.

**Comment by szetszwo** [comment]
@kerneltime , @duongkame , thanks a lot for reviewing this!

**Pull request merged**